v03_pipeline/lib/misc/io.py

```diff
@@ -147,16 +148,25 @@ def import_callset(
 def select_relevant_fields(
     mt: hl.MatrixTable,
     dataset_type: DatasetType,
+    additional_row_fields: None | list[str] = None,
 ) -> hl.MatrixTable:
     mt = mt.select_globals()
-    optional_row_fields = [
-        row_field
-        for row_field in dataset_type.optional_row_fields
-        if hasattr(mt, row_field)
-    ]
-    mt = mt.select_rows(*dataset_type.row_fields, *optional_row_fields)
-    mt = mt.select_cols(*dataset_type.col_fields)
-    return mt.select_entries(*dataset_type.entries_fields)
+    mt = mt.select_rows(
+        **{field: parse_nested_field(mt, field) for field in dataset_type.row_fields},
```

Comment:
This is minor but the code to create and unpack a dictionary of fields is repeated here so you could create a helper function that does this, takes arguments 'mt' and 'field_type' or something and returns 
```
**{field: parse_nested_field(mt, field) for field in dataset_type[field_type]}
```
where field_type could be row_fields, col_fields, or entries_fields

Reply:
yeah, I actually don't know on this one.  I'm pretty lukewarm on one line functions because of the additional layer of indirection but can see the case here.

My guess is there's some better deduplication that can happen with the reference data selection [here](https://github.com/broadinstitute/seqr-loading-pipelines/pull/797/files/1172bf69c49bd71077309c293cbb9bbca27d46d1#diff-8e147d1242ec201292bdadc21e13034e1e0e4ce60ccccb053f1833ee3abd02fdR112).  I didn't go as in-depth as I could have on unification.